rasa/nlu/extractors/entity_synonyms.py

```diff
@@ -39,23 +54,50 @@ def prepare_partial_training(
 
         See parent class for more information.
         """
-        for key, value in list(training_data.entity_synonyms.items()):
-            self.add_entities_if_synonyms(key, value)
+        self._add_synonyms_from_data(training_data)
+
+    def train(
+        self,
+        training_data: TrainingData,
+        config: Optional[RasaNLUModelConfig] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Train this component."""
+        self._add_synonyms_from_data(training_data)
+        self._process_entity_examples(training_data.entity_synonyms)
 
     def train_chunk(
         self,
-        training_data_chunk: TrainingDataChunk,
+        data_chunk_files: List[DataChunkFile],
         config: Optional[RasaNLUModelConfig] = None,
         **kwargs: Any,
     ) -> None:
-        """Train this component on the given chunk.
+        """Trains this component using the list of data chunk files.
 
         See parent class for more information.
         """
-        for example in training_data_chunk.entity_examples:
+        for data_chunk in data_chunk_files:
+            training_data_chunk = TrainingDataChunk.load_chunk(data_chunk.file_path)
```

Comment:
This means we are loading the same files multiple times right? As we also do this is the classifier?

Reply:
yes, see comment above

Reply:
Ah no, did not see that this referred to entity synonyms. In this case each file is just loaded once.